### ISSUE TITLE
S3 integration for the registry should not require credentials

### ISSUE THREAD

**Issue by etsauer** [openshift/openshift-ansible#4222]
#### Description

The S3 integration for the registry currently requires the following variables be defined:

```
openshift_hosted_registry_storage_s3_accesskey=aws_access_key_id
openshift_hosted_registry_storage_s3_secretkey=aws_secret_access_key
```

However, this is not always required. 

For example, assuming my ec2 instances running the nodes have the proper IAM policy assigned to them (https://docs.docker.com/registry/storage-drivers/s3/#s3-permission-scopes), I can simply configure `region` and `bucket`, like the below:

```
version: 0.1
log:
  level: debug
http:
  addr: :5000
storage:
  cache:
    blobdescriptor: inmemory
  delete:
    enabled: true
  s3:
    region: us-east-1
    bucket: openshift-registry
auth:
  openshift:
    realm: openshift
middleware:
  registry:
    - name: openshift
  repository:
    - name: openshift
  storage:
    - name: openshift
```

As such, we should remove the requirement for the `s3_accesskey` and `s3_secretkey`.

**Comment by sdodson** [comment]
@etsauer @miminar We've verified that this works in the version of docker-distribution that's shipped in 3.5?

**Comment by miminar** [comment]
@sdodson just by looking at the code, `region` and `bucket` are the only mandatory parameters.
